packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -770,9 +770,11 @@ export class TypeResolver {
         .replace(/,/g, '.')
         .replace(/'([^']*)'/g, '$1')
         .replace(/"([^"]*)"/g, '$1')
+        .replace(/`([^`]*)`/g, '$1')
```

Comment:
This concern is valid for both this and the other escape, but I think we can construct scenarios where this could lead to a clash (2 different references ending up with the same name), right?

Reply:
Yes it could, and whenever char are replaced or removed.

l TS Type | Escaped value | Data group |
|---|---|---|
|`Omit<MyType, "_foo">`|`Omit_MyType._foo_`|The type `MyType` minus the key `_foo` (1)|
|`Omit<MyType, '_foo'>`|`Omit_MyType._foo_`|The type `MyType` minus the key `_foo` (1)|
|``Omit<MyType, `_foo`>``|`Omit_MyType._foo_`|The type `MyType` minus the key `_foo` (1)|
|``Omit<MyType, `${foo}`>``|`Omit_MyType._foo_`|The type `MyType` minus the key of type `foo` (2)|
|``Omit<MyType, `${fo}o`>``|`Omit_MyType._foo_`|The type `MyType` minus the key of (type `fo` + text `o`)  (3)|
|``Omit<MyType, `${"foo"}`>``|`Omit_MyType._foo_`|The type `MyType` minus the key `foo` (4)|
|_many more possibilities_|||

For the first 3, they are the same data, so it's not a big deal.

---

There is maybe another case (I don't tested it yet): classes, interfaces, types, enums are not "unique".
In TSOA, `refType` don't mention their source file, so:
```ts
// fileA.ts
import { MyType } from "./somewhere/in/the/project"
```
```ts
// fileB.ts
import { MyType } from "./in/another/place"
```
will be declared as the same schema in the OpenAPI file (to be confirmed), and so create a clash

Reply:
Can we find a way to rewrite that so it does not clash with other, but different type names?